### DISCUSSION TITLE
Global strain method with input mesh with tetrahedra elements giving zero solution despite of applied stress

### DISCUSSION THREAD

**Discussion by ngrilli** [idaholab/moose#24428] [Q&A Modules: Solid mechanics]
I tried to apply the global strain method with periodic boundary conditions based on the paper by @SudiptaBiswas

https://www.sciencedirect.com/science/article/pii/S0168874X20301165

I have an .inp mesh file and I copied the input file example global_strain_uniaxial.i
from the tensor mechanics module.

The example works well of course, and the only change I made is using the FileMeshGenerator at the beginning
instead of GeneratedMeshGenerator.
Then I define periodic boundary conditions using the node sets in the .inp file and translation vectors
instead of the original auto_direction.

Relevant input files are here:

https://github.com/ngrilli/c_pfor_am/tree/main/test/tests/GlobalStrain 

When I run the simulation I get a result with zero displacements and there is no deformation
even though I would expect it because of the applied stress.

What could be the problem?

The only differences I can notice with respect to the working examples are:

1) I am using linear tetrahedra elements instead of hexahedra.

2) The nodes on my periodic surfaces do not match spatially, but I thought MOOSE has the capability to handle this
and I am not getting any error anyway.

Thank you very much in advance for your help,
Best Regards,
Nicolò

**Comment by hugary1995** [comment]
PBC nodesets have to be conformal. There has been some recent development on weak enforcement of PBC. Though I'm not sure what's the current status of that. @lindsayad

**Reply by ngrilli** [reply]
@hugary1995  thank you very much for the answer. I will try matching the nodes on the periodic surfaces.
Indeed I am very interested in using weak enforcements if that allows me to progress with the current mesh.
